src/developer-documentation/ibc/ibc.md

```diff
@@ -0,0 +1,999 @@
+    #[ink(message)]
+    fn transfer(&mut self, to: AccountId, id: Id, value: Balance, data: Vec<u8>) -> Result<(), PSP37Error>;
+
+    /// Transfers `amount` tokens of token type `id` from `from` to `to`. Also some `data` can be passed.
+    ///
+    /// On success a `TransferSingle` event is emitted.
+    ///
+    /// # Errors
+    ///
+    /// Returns `TransferToZeroAddress` error if recipient is zero account.
+    ///
+    /// Returns `NotAllowed` error if transfer is not approved.
+    ///
+    /// Returns `InsufficientBalance` error if `from` doesn't contain enough balance.
+    ///
+    /// Returns `SafeTransferCheckFailed` error if `to` doesn't accept transfer.
+    #[ink(message)]
+    fn transfer_from(
+        &mut self,
+        from: AccountId,
+        to: AccountId,
+        id: Id,
+        amount: Balance,
+        data: Vec<u8>,
+    ) -> Result<(), PSP37Error>;
+}
+```
+
+## 2.2 struct
+```js
+    pub struct Addr(String);
+
+    pub struct MessageInfo {
+        pub sender: Addr,
+        pub funds: Vec<Coin>,
+    }
+
+    pub struct Coin {
+        pub denom: String,
+        pub amount: u128,
+    }
+
+    pub struct Cw20Coin {
+        pub address: String,
+        pub amount: u128,
+    }
+
+    pub struct Cw20ReceiveMsg {
+        pub sender: String,
+        pub amount: u128,
+        pub msg: Vec<u8>,
+    }
+
+    pub struct AllowMsg {
+        pub contract: String,
+        pub gas_limit: Option<u64>,
+    }
+
+    pub struct InitMsg {
+        /// Default timeout for ics20 packets, specified in seconds
+        pub default_timeout: u64,
+        /// who can allow more contracts
+        pub gov_contract: String,
+        /// initial allowlist - all cw20 tokens we will send must be previously allowed by governance
+        pub allowlist: Vec<AllowMsg>,
+        /// If set, contracts off the allowlist will run with this gas limit.
+        /// If unset, will refuse to accept any contract off the allow list.
+        pub default_gas_limit: Option<u64>,
+    }
+
+    pub enum TimeoutHeight {
+    #[codec(index = 0)]
+    Never,
+    #[codec(index = 1)]
+    At(runtime_types::ibc::core::ics02_client::height::Height),
+    }
+
+    pub struct TransferMsg {
+        /// The local channel to send the packets on
+        pub channel: String,
+        /// The remote address to send to.
+        /// Don't use HumanAddress as this will likely have a different Bech32 prefix than we use
+        /// and cannot be validated locally
+        pub remote_address: String,
+        /// How long the packet lives in seconds. If not specified, use default_timeout
+        pub timeout: Option<u64>,
+        /// An optional memo to add to the IBC transfer
+        pub memo: Option<String>,
+    }
+
+
+    pub enum Amount {
+        Native(Coin),
+        // FIXME? USe Cw20CoinVerified, and validate cw20 addresses
+        Cw20(Cw20Coin),
+    }
+
+    impl Amount {
+        // TODO: write test for this
```

Comment:
TODO looks fine in code, not sure about docs.

Reply:
ok, i will remove it